components/sup/src/manager/mod.rs

```diff
@@ -1015,7 +1017,11 @@ where
             .as_ref(),
     ) {
         Ok(mut file) => {
-            match write!(&mut file, "{}", process::current_pid()) {
+            let pid = match env::var(launcher_protocol::LAUNCHER_PID_ENV) {
```

Comment:
This is a really clever way to get this done. Very well thought out IMO to the point where the Supervisor can still be started by an older launcher